From d4514ccc64c2af7264cb0d16d8c64d4ec10fbeb3 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Mon, 3 Nov 2014 07:52:31 -0500
Subject: [infiniband] iser: Clarify a duplicate counters check

Message-id: <319900043c26d73cdfffe6013929d78ac3c084d6.1415000301.git.avadai@redhat.com>
Patchwork-id: 98702
O-Subject: [Patch RHEL7.1 33/37] IB/iser: Clarify a duplicate counters check
Bugzilla: 1107622
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Doug Ledford <dledford@redhat.com>

From: Roi Dayan <roid@mellanox.com>

This is to prevent someone from thinking that this code section is
redundant.

Signed-off-by: Ariel Nahum <arieln@mellanox.com>
Signed-off-by: Roi Dayan <roid@mellanox.com>
Signed-off-by: Or Gerlitz <ogerlitz@mellanox.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 8d4aca7f04a098086c6545df158e5a022cbbfcd1)
Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/ulp/iser/iser_verbs.c b/drivers/infiniband/ulp/iser/iser_verbs.c
index 06a49b3..3ef167f 100644
--- a/drivers/infiniband/ulp/iser/iser_verbs.c
+++ b/drivers/infiniband/ulp/iser/iser_verbs.c
@@ -783,7 +783,10 @@ static void iser_disconnected_handler(struct rdma_cm_id *cma_id)
    iser_err("iscsi_iser connection isn't bound\n");
  }
 
- /* Complete the termination process if no posts are pending */
+ /* Complete the termination process if no posts are pending. This code
+  * block also exists in iser_handle_comp_error(), but it is needed here
+  * for cases of no flushes at all, e.g. discovery over rdma.
+  */
  if (ib_conn->post_recv_buf_count == 0 &&
      (atomic_read(&ib_conn->post_send_buf_count) == 0)) {
   complete(&ib_conn->flush_completion);
-- 
1.7.1